### ISSUE TITLE
Triller playback is in wrong order

### ISSUE THREAD

**Issue by lgmventura** [BernardGreenberg/MuseScorePlugins#1]
After applying the triller, the triller starts from the note above and not from the note that have _tr_ on it.
Steps to reproduce:
1) Insert a note, e.g. C.
2) Put a _tr_ on it.
3) Select the note and open the plugin _triller_.
4) Leave the configuration as it is or change it the way you want. Let's suppose you leave one tone upwards.
5) Apply.
What you expect:
a triller starting from C followed by D, C, D, C, …
What you get:
a triller starting from D followed by C, D, C, D, …

**Comment by BernardGreenberg** [comment]
Yup; this is a current limitation. I've upgraded the documentation. It already says "Baroque" trills, which in MuseScore means "starting on the upper".  I plan to add a "starting on the the lower" feature, soon. Keep watch on CHANGES.md.  Completely legitimate need, thanks.

**Comment by lgmventura** [comment]
Thanks a lot! I didn't know what they mean with "baroque" trills.
I think this is the part in the QML file that can be changed:
`        if (vor_von_oben) {
            ppush([oben_steps, 0, unten_steps, 0]);
        } else if (vor_von_unten){
            ppush([unten_steps, 0]);
        }`

So, adding an `if starting_on_the_lower`:
`        if (vor_von_oben) {
            ppush([0, oben_steps, 0, unten_steps]);
        } else if (vor_von_unten){
            ppush([0, unten_steps]);
        }`

I tried to change this in that QML file, but it had no effect. I think one has to compile to that other QMLC file…

**Comment by BernardGreenberg** [comment]
Feel free to play around with the code, but it needs real UI, including ruling out the Vorschlag options and handling the end-matching - an odd number of beats are needed, and the end-step meaning changes.  Enjoy yourself, but doing it "right" involves all those considerations, and I promise I will. Delete the QMLC; it compiles it itself.  Better yet, develop from the Plugin Creator (in the Plugin menu) designed for that purpose.

**Comment by lgmventura** [comment]
Thank you! Great, I could solve it creating a checkbox for "baroque". Now, it's possible to choose either of the two ways. It seems to work, but I still haven't tested all cases. How can I create a pull-request here?

**Comment by BernardGreenberg** [comment]
Didn't think of that (let you do the work!) What a great idea.  The checkboxes should be "Baroque" "Standard", or maybe better:

Start on [ ]lower [ ] upper
default if nothing there, upper, otherwise you have to decode the trill which is there, as it does now, and decide which case it is.  Do you have a branch, so I can look over your code?

**Comment by lgmventura** [comment]
Great! I used "baroque" to keep the MuseScore terminology, but it can be changed. I'll create a branch tomorrow and upload the file. ;)

**Comment by BernardGreenberg** [comment]
Did you upgrade the "figure out what's already there" code to identify start-lower trillls and count the beats properly?

**Comment by lgmventura** [comment]
I wanted to create a branch, but I'm not a collaborator in this repository. Regarding your question, I still didn't, I was thinking about it. In the current state, if you call the plugins, it figures out what's already there, except for this. It leaves "baroque" marked by default. But I'll have a look if I can update this as well, I haven't tried yet.

**Comment by BernardGreenberg** [comment]
Do you know how to add a collaborator? I'll make you be one if you tell me.  Try to understand and upgrade the "figure out what we have" code ...

**Comment by BernardGreenberg** [comment]
You are invited now. Please understand and upgrade the "what we have" code.  The case of seeing a MuseScore start-lower trill will be common.

**Comment by lgmventura** [comment]
It doesn't seem that I can create a branch or pull request. Could you please check if I was correctly invited?

**Comment by BernardGreenberg** [comment]
yeah, "Awaiting Igmventura's response" ...

**Comment by lgmventura** [comment]
Ok, I wasn't notificated and I'm still a beginner on GitHub, I think I have to fork it before…

**Comment by BernardGreenberg** [comment]
I'm not a GitHub wizard, either, but you can probably find out via websearching how to get this right. At worst, you can email me your code.

**Comment by lgmventura** [comment]
Yeah, I updated the "what we have" code. Also need to update the comments saying that it would lie about that hahah. I'm still going to try to create a branch/PR.

**Comment by lgmventura** [comment]
Wait, sorry, actually it's not done for the other cases, like with vorschlag and nachschlag. Working on this now…

**Comment by BernardGreenberg** [comment]
The Vorschlag options are not meaningful for start-lower trills -- the option specifies a specific beginning.  "start lower with Vorschlag" cannot be asked for or recognized.  As for the Nachschlag mordent, that's still definitely possible.

**Comment by BernardGreenberg** [comment]
There are 4 mutually-exclusive possible starts -- straight upper, straight lower, Vorschlag von oben and Vorschlag von unten

**Comment by lgmventura** [comment]
Thank you, I was facing exactly that question, if it's possible to combine start-lower with Vorschlag. But my knowledge in music notation is still limited to answer this. Now I know I can ignore this and print a console message.

**Comment by BernardGreenberg** [comment]
Either merge the upper/lower with the Vorschlag options into a new one-of-4 UI or make sure Vorschlag and lower disable each other.  I think the one-of-4 option is the best.

**Comment by lgmventura** [comment]
I've just done a first commit and push to a new branch in my forked repository. I'm keeping the name "baroque" because "start from upper/lower" might lead to confusion with the "Vorschläge". And it's also consistent with the MS terminology. I still have to do these UI changes, but it's already usable.

**Comment by BernardGreenberg** [comment]
Can you post here a pointer to the latter?I'd love to see it, try it, and examine/critique the code ...  I really, really do thank you for taking this upon yourself!

**Comment by lgmventura** [comment]
Did you find it here? https://github.com/lgmventura/MuseScorePlugins/tree/ms-non-baroque

**Comment by BernardGreenberg** [comment]
Thanks, I am studying this.  I'm not crazy about the non-Baroque default, an incompatible change, but I guess that's MuseScore's default, too. I wish that could be set per-user.
________________________________

**Comment by BernardGreenberg** [comment]
Could you figure out how to save and restore the default choice?  The tempo change plugin has (or had) a way of using the QML profile mechanism to save and restore values across invocations. I'm willing to live with a "default default" of "from lower" if my last setting can be remembered and used as the running default.
